scheduler/task.go

```diff
@@ -42,6 +42,57 @@ func createTaskInfo(task types.EremeticTask, offer *mesos.Offer) (types.Eremetic
 		})
 	}
 
+	var portResources []*mesos.Value_Range
+	var portMapping   []*mesos.ContainerInfo_DockerInfo_PortMapping
+
+	if(len(task.Ports) > 0) {
```

Comment:
This is a quite big block of code to inline in this function we should try to break it up.

For an initial suggestion it could be broken down into these steps (which each could be their own function)
* a) Get N ports from an offer
* b) Construct portMappings from the ports definition and a list of ports
* c) Construct ranges resource of ports from list of ports

as an added bonus a could also be reused to implement the matcher